java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -203,6 +195,44 @@ class DatabaseImpl(
         )
     }
 
+    private fun getEntityFieldValue(
+        fieldValueId: FieldValueId,
+        field: SchemaField,
+        db: SQLiteDatabase,
+        counters: Counters?
+    ): Any = when {
+        field.isCollection -> {
+            counters?.increment(DatabaseCounters.GET_ENTITY_FIELD_VALUE_COLLECTION)
+            getCollectionEntries(fieldValueId, field.typeId, db, counters)
+        }
+        isPrimitiveType(field.typeId) -> {
+            counters?.increment(DatabaseCounters.GET_ENTITY_FIELD_VALUE_PRIMITIVE)
+            getPrimitiveValue(fieldValueId, field.typeId, db, counters)
+        }
+        else -> {
+            counters?.increment(DatabaseCounters.GET_ENTITY_FIELD_VALUE_REFERENCE)
+            getReferenceValue(fieldValueId, db)
+        }
+    }
+
+    private fun getReferenceValue(
+        entityRefId: FieldValueId,
+        db: SQLiteDatabase
+    ): Any = db.rawQuery(
```

Comment:
Rather than return `Any`, it might make more sense to have each of the `getCollectionEntries`, `getPrimitiveValue` and `getReferenceValue` return "Referencable".

Reply:
I might tackle this in a follow up PR